### ISSUE TITLE
Analyzer: Impossible to query system.zookeeper

### ISSUE THREAD

**Issue by jrdi** [ClickHouse/ClickHouse#47255]
Queries to `system.zookeeper` with `allow_experimental_analyzer = 1` are failing as follows:

```sql
SELECT *
FROM system.zookeeper
WHERE path = 'path'
SETTINGS allow_experimental_analyzer = 1

Query id: 24da186e-f8ab-437f-822e-40430cc2f3ae

0 rows in set. Elapsed: 0.187 sec.

Received exception from server (version 23.2.1):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: SELECT from system.zookeeper table must contain condition like path = 'path' or path IN ('path1','path2'...) or path IN (subquery) in WHERE clause unless `set allow_unrestricted_reads_from_keeper = 'true'`.. (BAD_ARGUMENTS)
When deactivating the analyzer everything works fine:
```

Same query with `allow_experimental_analyzer = 0` is working fine:
```sql
SELECT *
FROM system.zookeeper
WHERE path = 'path'
SETTINGS allow_experimental_analyzer = 0

Query id: 38167da5-23f8-4cd5-8be6-55a2b71fbfc1

Ok.

0 rows in set. Elapsed: 0.002 sec.
```

This is only happening in master, CH 23.2 is allowed to process those queries just fine.

Relates to #42648

**Comment by novikd** [comment]
I suppose it to be fixed in #48485

**Comment by jrdi** [comment]
Fixed in https://github.com/ClickHouse/ClickHouse/pull/48485
